src/main/java/com/rsakin/vocid22tracker/controller/VocidTrackerController.java

```diff
@@ -0,0 +1,107 @@
+package com.rsakin.vocid22tracker.controller;
+
+import com.rsakin.vocid22tracker.model.dto.PatientDTO;
+import com.rsakin.vocid22tracker.model.entity.Patient;
+import com.rsakin.vocid22tracker.service.VocidTrackerService;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import java.util.List;
+
+@Slf4j
+@Validated
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/vocid22")
+public class VocidTrackerController {
+
+    private final VocidTrackerService vocidTrackerService;
+
+    @ApiOperation("Get all infected Patients")
+    @GetMapping(value = "/patient/all")
+    public ResponseEntity<List<Patient>> getAllPatients() {
+        List<Patient> allPatients = vocidTrackerService.getAllPatients();
+        return new ResponseEntity<>(allPatients, HttpStatus.OK);
+    }
+
+    @ApiOperation("Get any already defined patient by id")
+    @GetMapping(value = "/patient/{patient_id}")
+    public ResponseEntity<Patient> getPatient(
+            @PathVariable @Min(1) final Long patient_id
+    ) {
+        Patient patient = vocidTrackerService.getPatientById(patient_id);
+        return new ResponseEntity<>(patient, HttpStatus.OK);
+    }
+
+    @ApiOperation("Create a patient")
+    @PostMapping(value = "/patient/create")
```

Comment:
Would've been nice to see a more REST styled naming pattern with nouns to represent resources rather than RPC style invocations.
e.g 

- `GET /patients` to get all patients
- `GET /patients/{patient-id}` to get a specific patient
- `POST /patients` to create a new patient entry
etc..

Reply:
Yea, sometimes I wrote like this sometimes I wrote as you mentioned, I guess it depends on my mood :D and I had a bootcamp  at Sunday and I presented some these kind of urls to students and probably I also did like it :)